### PR TITLE
refactor(Card): convert to typescript

### LINKED ISSUE #5197: Add Typescript types to Card, CardFooter, CardHeader
As we add Typescript support, want to call out a few helpful tips from the Carbon TypeScript guide that is largely relevant to our library as well.

### Changing the extension via git

When changing the file extension — **do not simply rename the file**. **Instead, change the extension via `git`.**

For example,

```
git mv packages/ibm-products/src/components/ComponentName/ComponentName.js packages/ibm-products/src/components/ComponentName/ComponentName.tsx
```

### Do not add types to internal components or functions
- Leave internals as `.js`
- Add a [jsdoc type annotation](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#providing-type-hints-in-js-via-jsdoc) of `/** @type any */`
- [Enable errors](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#ts-check) by adding `// @ts-check` to the first line in the file.

### Testing changes
A few different options for testing changes —

- At the bottom of a file, write a dummy component that uses the component that you’re converting to ensure that you can still pass all appropriate props.
- Take a storybook example for a component, copy and paste it into the bottom of the `.tsx` file and validate whether or not it accepts the props as you’ve defined them.

```[tasklist]
### Tasks
- [ ] Change extension to `.tsx` **via git** (**do not simply rename**, see above)
- [ ] Copy the PropTypes definitions above the component definition
- [ ] Retool the PropTypes to be a TypeScript interface
- [ ] Fix errors as they appear
- [ ] Do NOT add types to internal components or functions that are not exported as part of the public API
- [ ] Test your changes (see above)
```

### Resources
Carbon’s guide on [adding component types](https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md)

### PR DESCRIPTION
**Pull request by anamikaanu96**
Closes #5197 

Add Typescript types to Card, CardFooter, CardHeader

#### What did you change? 
Converted below files : 
packages/ibm-products/src/components/Card/Card.js to packages/ibm-products/src/components/Card/Card.tsx
packages/ibm-products/src/components/Card/CardHeader.js to packages/ibm-products/src/components/Card/CardHeader.tsx
packages/ibm-products/src/components/Card/CardFooter.js to packages/ibm-products/src/components/Card/CardFooter.tsx

### CODE REVIEW

**Review by amal-k-joy — CHANGES_REQUESTED** [review]

**Review by matthewgallo — CHANGES_REQUESTED** [review]
There seems to be a type error with `footerActionIcon`

https://github.com/carbon-design-system/ibm-products/actions/runs/10164871770/job/28111339889?pr=5760

File: packages/ibm-products/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -22,67 +22,136 @@ import { pkg } from '../../settings';
...
-  clickZone: 'one',
-  mediaPosition: 'top',
-  overflowActions: Object.freeze([]),
-  primaryButtonKind: 'primary',
-  primaryButtonPlacement: 'bottom',
-  productive: false,
-  secondaryButtonKind: 'secondary',
-  secondaryButtonPlacement: 'bottom',
-  titleSize: 'default',
+// const defaults = {
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/Card/Card.tsx:25]: can you remove these commented out defaults?

**Review by davidmenendez — CHANGES_REQUESTED** [review]
LGTM just one small comment

**Review by davidmenendez — DISMISSED** [review]
LGTM

File: packages/ibm-products/src/components/Card/Card.tsx [hunk 2]
```diff
@@ -21,68 +21,122 @@ import { CardFooter } from './CardFooter';
...
+  sequence?: number;
+
+  /**
+   * **Experimental?** For all cases a `Slug` component can be provided.
+   * Clickable tiles only accept a boolean value of true and display a hollow slug.
+   */
+  slug?: ReactNode | boolean;
+
+  status?: 'complete' | 'incomplete';
+  title?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/Card/Card.tsx:89]:
```suggestion
  title?: ReactNode;
```
ReactNode supports string, no need to add it here

File: packages/ibm-products/src/components/Card/Card.tsx [hunk 3]
```diff
@@ -21,68 +21,122 @@ import { CardFooter } from './CardFooter';
...
+  onClick?: () => void;
+  href?: string;
+};
+interface CardProp extends PropsWithChildren {
+  actionIcons?: readonly ActionIcon[];
+  actionsPlacement?: 'top' | 'bottom';
+  children?: ReactNode;
+  className?: string;
+  clickZone?: 'one' | 'two' | 'three';
+  description?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/Card/Card.tsx:52]:
```suggestion
  description?: ReactNode;
```

File: packages/ibm-products/src/components/Card/CardHeader.tsx [hunk 4]
```diff
@@ -17,7 +17,39 @@ const defaults = {
   titleSize: 'default',
 };
 
-export let CardHeader = ({
+interface CardHeaderProps {
+  actions?: ReactNode[] | ReactNode;
+  description?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/Card/CardHeader.tsx:22]:
```suggestion
  description?: ReactNode;
```

File: packages/ibm-products/src/components/Card/CardHeader.tsx [hunk 5]
```diff
@@ -17,7 +17,39 @@ const defaults = {
...
+  hasActions?: boolean;
+  /**
+   * is the host card clickable
+   */
+  inClickableCard?: boolean;
+  label?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/Card/CardHeader.tsx:28]:
```suggestion
  label?: ReactNode;
```

File: packages/ibm-products/src/components/Card/CardHeader.tsx [hunk 6]
```diff
@@ -17,7 +17,39 @@ const defaults = {
...
+  secondaryButtonKind?: 'secondary' | 'ghost';
+  secondaryButtonPlacement?: 'top' | 'bottom';
+  secondaryButtonText?: string;
+  /**
+   * **Experimental:** For all cases a `Slug` component can be provided.
+   * Clickable tiles only accept a boolean value of true and display a hollow slug.
+   */
+  slug?: ReactNode;
+
+  title?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/Card/CardHeader.tsx:48]:
```suggestion
  title?: ReactNode;
```

File: packages/ibm-products/src/components/GetStartedCard/GetStartedCard.tsx [hunk 7]
```diff
@@ -35,12 +34,12 @@ interface GetStartedCardProps extends PropsWithChildren {
...
    * Provides the action icon that's displayed at the footer of the card
    */
-  footerActionIcon: () => void | object;
+  footerActionIcon: React.ElementType;
 
   /**
    * Optional label for the top of the card
    */
-  label?: string | object | ReactNode;
+  label?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/GetStartedCard/GetStartedCard.tsx:42]:
```suggestion
  label?: ReactNode;
```

File: packages/ibm-products/src/components/GetStartedCard/GetStartedCard.tsx [hunk 8]
```diff
@@ -75,7 +74,7 @@ interface GetStartedCardProps extends PropsWithChildren {
   /**
    * Title that's displayed at the top of the card
    */
-  title?: string | object | ReactNode;
+  title?: string | ReactNode;
```
Comment (AlexanderMelox) [inline packages/ibm-products/src/components/GetStartedCard/GetStartedCard.tsx:77]:
```suggestion
  title?: ReactNode;
```

**Review by AlexanderMelox — CHANGES_REQUESTED** [review]

**Review by amal-k-joy — APPROVED** [review]

**Review by AlexanderMelox — APPROVED** [review]

**Review by matthewgallo — APPROVED** [review]

**Pull request merged**